usr/bin/cylc

```diff
@@ -16,54 +16,156 @@
+# select them using CYLC_HOME in your .bashrc.
+#
+# Create a new venv and install cylc-flow:
+#   $ cd <my-cylc-flow-clone>
+#   $ python -m venv venv
+#   $ . venv/bin/activate
+#   $ pip install -e . 
+#
+# Set CYLC_HOME in .bashrc to select this venv (see explanation above):
+#
+# Note you can also `pip install` cylc-flow releases under a common root
+# location for CYLC_VERSION based selection, like for Conda:
+#   $ CYLC_HOME_ROOT=$HOME/cylc-flow/releases  # (set in this wrapper too)
+#   $ python -m venv $CYLC_HOME_ROOT/cylc-8.0a2
+#   $ . $CYLC_HOME_ROOT/cylc-8.0a2/bin/activate
+#   $ python -m pip install cylc-flow==8.0a2
+# But there is no need to do this if you also have full system releases (as
+# opposed to cylc-flow only) installed by Conda.
+#
+# INSTALLING LEGACY cylc 7 RELEASE TARBALLS BY HAND
+#--------------------------------------------------
+# cylc-flow release tarballs now unpack to (e.g.) cylc-flow-7.9.1. To work with
+# this wrapper the directory should be renamed to "cylc-7.9.1". Then follow
+# version-specific installation instructions. Running "make" should create a
+# file called VERSION that contains just the version string (e.g.) "7.9.1".
+#
+# INSTRUCTIONS FOR USERS
+#-----------------------
+# + Set CYLC_HOME_ROOT_ALT in .bashrc to point any local conda releases, e.g.:
+#     $ export CYLC_HOME_ROOT_ALT=$HOME/miniconda3/envs
+#   (Environments must be named for the cylc version, e.g. "cylc-8.0.2")
+# + Set CYLC_VERSION e.g. "8.0.0" to select a version in the root locations
+#   - CYLC_VERSION is propagated to task jobs by the scheduler. If you set it
+#   in .bashrc it must default to an existing value to avoid overiding jobs:
+#       $ export CYLC_VERSION=${CYLC_VERSION:-8.0.0}
+#   - Do not explicitly select the default "cylc" symlink as a version
+# + Set CYLC_HOME in .bashrc (on scheduler and job hosts) to select a specific
+#   version outside of the ROOT location
+#   - CYLC_HOME should point to a Cylc 8 venv, or a plain Cylc 7 directory. It
+#     must be set in .bashrc so that task jobs see it too
+# + Any of these settings in .bashrc must be replicated on job hosts.
+#
+##############################!!! EDIT ME !!!##################################
+# Centrally installed Cylc releases:
+CYLC_HOME_ROOT="${CYLC_HOME_ROOT:-/opt}"
 
-CYLC_HOME_ROOT=${CYLC_HOME_ROOT:-/opt} # !!! EDIT ME !!!
+# Note users can set CYLC_HOME_ROOT_ALT as well (see above), e.g.:
+# CYLC_HOME_ROOT_ALT=${HOME}/miniconda3/envs
+###############################################################################
 
-if [[ -z ${CYLC_HOME:-} ]]; then
-    if [[ -n ${CYLC_VERSION:-} && -d $CYLC_HOME_ROOT/cylc-$CYLC_VERSION ]]; then
-        CYLC_HOME=$CYLC_HOME_ROOT/cylc-$CYLC_VERSION
-    else
-        CYLC_HOME=$CYLC_HOME_ROOT/cylc
+# If CYLC_HOME is not defined try CYLC_HOME_ROOT/cylc-CYLC_VERSION
+if [[ -z "${CYLC_HOME}" ]]; then
+    if [[ -z ${CYLC_VERSION} ]]; then
+        echo 1>&2 "ERROR: CYLC_HOME or CYLC_VERSION must be defined"
```

Comment:
This no longer supports the symlink default version?

Reply:
damn-it_2

Reply:
fixed (late night, hope I didn't screw it up again!) :wave: